projects/plugins/social/src/js/editor.js

```diff
@@ -27,27 +30,37 @@ domReady( () => {
 } );
 
 registerPlugin( 'jetpack-social', {
-	render: () => (
+	render: () => <JetpackSocialSidebar />,
+} );
+
+const JetpackSocialSidebar = () => {
+	const [ isModalOpened, setIsModalOpened ] = useState( false );
+
+	const openModal = useCallback( () => setIsModalOpened( true ), [] );
+	const closeModal = useCallback( () => setIsModalOpened( false ), [] );
```

Comment:
I'm not sure that `useCallback` will give us any gain here. [This post suggests](https://kentcdodds.com/blog/usememo-and-usecallback) it could actually be worse!

Reply:
ESLint was yelling at me without it—I'm happy to switch to inline arrow functions if possible, that's my usual preference.

Reply:
I suppose that might be to safeguard against it being used in a dependency array or something in the component. It might be worth discussing that rule, it seems debatable.

The equivalent in the `SocialPreviews` component is inline. That must have been written before the rule was enforced.